File: 03 . Java_Programming/12.Single.java
class Base                // 8 byte
{
    public int i;
    public int j;

    public void Fun()
    {
        System.out.println("Inside Fun \n");
    }
}

class Derived extends Base                 // 12 byte
{
    public int x;

    public void Gun()
    {
        System.out.println("Inside Gun \n");
    }
}

class Single
{
    public static void main(String A[])
    {
        Base bobj = new Base();
        Derived dobj = new Derived();

        bobj.Fun();

        dobj.Fun();
        dobj.Gun();
        
    }
}

File: 02 . C++ _Programming/12.Single.cpp
#include <iostream>

using namespace std;

class Base
{
    public:
        int i,j;

        void Fun()
        {
            cout<<"Inside Base Fun\n";
        }        

};

class Derived : public Base
{
    public:
        int x;

        void Gun()
        {
            cout<<"Inside Derived Gun\n";
        }
};

int main()
{
    Base bobj;
    Derived dobj;

    cout<<"Size of Base class object :"<<sizeof(bobj)<<"\n";
    cout<<"Size of Base class object :"<<sizeof(dobj)<<"\n";

    return 0;
}